docker/Dockerfile

```diff
@@ -24,15 +24,24 @@
 #     -t the target repo and tag name
 #     more options can be found with -h
 
-ARG BASE_IMAGE=8-jre-slim
-FROM openjdk:${BASE_IMAGE}
+ARG BASE_IMAGE=openjdk:8-jre-slim
+ARG spark_provided="spark_builtin"
+ARG spark_home="/opt/spark"
```

Comment:
Does this ARG `spark_home` work? 

From the docker official documents:
> An ARG instruction goes out of scope at the end of the build stage where it was defined. To use an arg in multiple stages, each stage must include the ARG instruction.

FYI: https://docs.docker.com/engine/reference/builder/#scope

Reply:
Nice catch, i will check.